src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -702,6 +703,12 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                     final String name = ((NameCallback) callback).getDefaultName();
                     try {
                         setAuthenticationName(name);
+                        if (callback instanceof ExclusiveNameCallback) {
+                            final ExclusiveNameCallback exclusiveNameCallback = ((ExclusiveNameCallback) callback);
+                            if (exclusiveNameCallback.needsExclusiveAccess()) {
+                                exclusiveNameCallback.setExclusiveAccess(stateRef.get().getRealmIdentity().isLocked());
```

Comment:
Rather than having some realms support exclusive access and some not, I think it would be better to have a variation of setAuthenticationName() which has an 'exclusive' flag, which, when true, would use the for-update variation (assuming the realm is updateable) but fail otherwise.  Otherwise we'll be getting exclusive access when we don't need it (which we would not, for most mechanisms).